### ISSUE TITLE
Is this component compatible with IOS 11?

### ISSUE THREAD

**Issue by fanglinwei** [lkzhao/UIComponent#21]
Is this component compatible with IOS 11

**Comment by zzycami** [comment]
It use api of swiftui, ofcause it not compatible with IOS 11

**Comment by fanglinwei** [comment]
> It use api of swiftui, ofcause it not compatible with IOS 11

I analyzed its source code and built it with UIKit API

**Comment by zzycami** [comment]
Yes, this library is combine uikit with swiftui, this is this library is doing, but it contain swiftui , and swiftui is a library which is noly support ios 13 and above. so this library is not compatible with IOS 11. you can try to build this library with ios 11, you will get failed if you try.

**Comment by fanglinwei** [comment]
> Yes, this library is combine uikit with swiftui, this is this library is doing, but it contain swiftui , and swiftui is a library which is noly support ios 13 and above. so this library is not compatible with IOS 11. you can try to build this library with ios 11, you will get failed if you try.

It doesn't use swiftui features, just some IOS 13 features It can be supported. If it relies on swiftui, it can also do swiftui. What is the significance of its existence?

**Comment by fanglinwei** [comment]
> Yes, this library is combine uikit with swiftui, this is this library is doing, but it contain swiftui , and swiftui is a library which is noly support ios 13 and above. so this library is not compatible with IOS 11. you can try to build this library with ios 11, you will get failed if you try.

Wǒ bìng méiyǒu kàn dào tā importle swiftui de xiāngguān kuàngjià, zhǐshì shǐyòngle UIKit de yīxiē iOS13 zhīchí de tèxìng, zhèxiē shì kěyǐ zuò xiàng gèng dī de bǎnběn jiānróng de.

volume_up
70 / 5,000
翻译结果
I didn't see that it imported the relevant framework of swiftui, but just used some iOS13 supported features of UIKit, which can be compatible with lower versions.

**Comment by fanglinwei** [comment]
> Yes, this library is combine uikit with swiftui, this is this library is doing, but it contain swiftui , and swiftui is a library which is noly support ios 13 and above. so this library is not compatible with IOS 11. you can try to build this library with ios 11, you will get failed if you try.

I didn't see that it imported the relevant framework of swiftui, but just used some iOS13 supported features of UIKit, which can be compatible with lower versions.

**Comment by zzycami** [comment]
I checked all the code, It does not use swiftui api, but you need to make basetoolbox to compatible with ios11, It should compatible with IOS 11

**Comment by fanglinwei** [comment]
> basetoolbox

Yes, I have studied all the codes of basetoolbox. It can be compatible with IOS 11

**Comment by fanglinwei** [comment]
> I checked all the code, It does not use swiftui api, but you need to make basetoolbox to compatible with ios11, It should compatible with IOS 11

Yes, I have studied all the codes of basetoolbox. It can be compatible with IOS 11

**Comment by zzycami** [comment]
> > I checked all the code, It does not use swiftui api, but you need to make basetoolbox to compatible with ios11, It should compatible with IOS 11
> 
> Yes, I have studied all the codes of basetoolbox. It can be compatible with IOS 11

Yes, problem fixed!

**Comment by lkzhao** [comment]
You guys are correct that this project doesn't use SwiftUI. If you want to put up a PR supporting iOS11. I can merge it in as long as it doesn't remove any existing functionality.

**Comment by fanglinwei** [comment]
> 你们是对的，这个项目不使用 SwiftUI。如果你想建立一个支持 iOS11 的 PR。只要它不删除任何现有功能，我就可以合并它。

This seems a bit cumbersome, and the code of basetoolbox needs to be adapted to iOS11 first. I have a local demo that has done this, and I will submit some prs later.

**Comment by fanglinwei** [comment]
> You guys are correct that this project doesn't use SwiftUI. If you want to put up a PR supporting iOS11. I can merge it in as long as it doesn't remove any existing functionality.

I maintain an [Extensions](https://github.com/Swift-Developers/Component-Extensions-Package) is also a simple extension that includes basetoolbox to see if I need to switch from basetoolbox to Extensions

**Comment by fanglinwei** [comment]
> You guys are correct that this project doesn't use SwiftUI. If you want to put up a PR supporting iOS11. I can merge it in as long as it doesn't remove any existing functionality.

I submitted a pr but there is still a iOS 13 version limitation on the use of some, I don't know how to replace it  [PR22](https://github.com/lkzhao/UIComponent/pull/22)

**Comment by AugustRush** [comment]
似乎使用some关键字就只能支持iOS 13以上了，需要把some替换成any，这可能需要改一些实现逻辑。另外状态绑定不像SwiftUI使用起来那么简单，建议可以参考 SwiftWebUI / ComponentKit实现一下。
